tests/ImageSharp.Tests/IO/ChunkedMemoryStreamTests.cs

```diff
@@ -0,0 +1,371 @@
+
+                // Validate the destination parameter first.
+                Assert.Throws<ArgumentNullException>(Destination, () => memoryStream.CopyTo(destination: null, bufferSize: 0));
+                Assert.Throws<ArgumentNullException>(Destination, () => memoryStream.CopyTo(destination: null, bufferSize: -1));
+
+                // Then bufferSize.
+                Assert.Throws<ArgumentOutOfRangeException>(BufferSize, () => memoryStream.CopyTo(Stream.Null, bufferSize: 0)); // 0-length buffer doesn't make sense.
+                Assert.Throws<ArgumentOutOfRangeException>(BufferSize, () => memoryStream.CopyTo(Stream.Null, bufferSize: -1));
+            }
+
+            // After the Stream is disposed, we should fail on all CopyTos.
+            Assert.Throws<ArgumentOutOfRangeException>(BufferSize, () => memoryStream.CopyTo(Stream.Null, bufferSize: 0)); // Not before bufferSize is validated.
+            Assert.Throws<ArgumentOutOfRangeException>(BufferSize, () => memoryStream.CopyTo(Stream.Null, bufferSize: -1));
+
+            ChunkedMemoryStream disposedStream = memoryStream;
+
+            // We should throw first for the source being disposed...
+            Assert.Throws<ObjectDisposedException>(() => memoryStream.CopyTo(disposedStream, 1));
+
+            // Then for the destination being disposed.
+            memoryStream = new ChunkedMemoryStream(this.allocator);
+            Assert.Throws<ObjectDisposedException>(() => memoryStream.CopyTo(disposedStream, 1));
+            memoryStream.Dispose();
+        }
+
+        [Theory]
+        [MemberData(nameof(CopyToData))]
+        public void CopyTo(Stream source, byte[] expected)
+        {
+            using var destination = new ChunkedMemoryStream(this.allocator);
+            source.CopyTo(destination);
+            Assert.InRange(source.Position, source.Length, int.MaxValue); // Copying the data should have read to the end of the stream or stayed past the end.
+            Assert.Equal(expected, destination.ToArray());
+        }
+
+        public static TheoryData<string> GetAllTestImages()
+        {
+            IEnumerable<string> allImageFiles = Directory.EnumerateFiles(TestEnvironment.InputImagesDirectoryFullPath, "*.*", SearchOption.AllDirectories)
+                .Where(s => !s.EndsWith("txt", StringComparison.OrdinalIgnoreCase));
+            var result = new TheoryData<string>();
+            foreach (string path in allImageFiles)
+            {
+                result.Add(path);
+            }
+
+            return result;
+        }
+
+        [Theory]
+        [MemberData(nameof(GetAllTestImages))]
+        public void DecoderIntegrationTest(string testFileFullPath)
+        {
+            if (!TestEnvironment.Is64BitProcess)
+            {
+                return;
+            }
+
+            Image<Rgba32> expected;
+            try
+            {
+                expected = Image.Load<Rgba32>(testFileFullPath);
```

Comment:
I know it's my code but I would really like to replace `Image.Load` with `TestFile` stuff now to make the tests at least a bit faster by utilizing it's caching mechanism.  Would do it, but I'm on a train now, may try later, if you also lack time.